kubeshift/login.py

```diff
@@ -0,0 +1,94 @@
+"""Perform Authentication."""
+import requests
+import six.moves.urllib.parse as url_parse
+
+from kubeshift.exceptions import KubeAuthFailedError
+
+
+def _parse_implicit_token(uri):
+    fragment = url_parse.urlparse(uri).fragment
+    params = dict(url_parse.parse_qsl(fragment, keep_blank_values=True))
+
+    if 'error' in params:
+        return None, {
+            'error': params.get('error'),
+            'description': params.get('error_description'),
+            'uri': params.get('error_uri')
+        }
+
+    if 'token_type' not in params:
+        return None, {
+            'error': 'missing_token_type',
+            'description': 'Missing token type parameter.',
+            'uri': uri
+        }
+
+    token = params.get('access_token')
+    if not token:
+        return None, {
+            'error': 'missing_access_token',
+            'description': 'Missing access token parameter.',
+            'uri': uri
+        }
+
+    return token, None
+
+
+def _request_token(base_url, username, password):
+    auth = (username, password)
+    params = {
+        'response_type': 'token',
+        'client_id': 'openshift-challenging-client'
```

Comment:
Good catch, I know it works against Openshift, but I haven't tested directly against k8s